src/ucp/wireup/wireup_cm.c

```diff
@@ -1008,19 +1008,23 @@ ucp_ep_cm_server_create_connected(ucp_worker_h worker, unsigned ep_init_flags,
     if (conn_request->listener->accept_cb) {
         conn_request->ep = ep;
         ucp_listener_schedule_accept_cb(conn_request);
+        goto out;
     } else {
-        ucs_free(conn_request->remote_dev_addr);
-        ucs_free(conn_request);
+        goto out_free_request;
     }
 
-    *ep_p = ep;
-
-out:
-    return status;
-
 err_destroy_ep:
     ucp_ep_destroy_internal(ep);
-    goto out;
+out_free_request:
+    ucs_free(conn_request->remote_dev_addr);
+    ucs_free(conn_request);
+out:
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    *ep_p = ep;
```

Comment:
maybe move `*ep_p = ep` before line 1008, where we already know status==UCS_OK
and here, can just "return status"

Reply:
this is what I did first, but initializing out-parameter in the end looks simpler to me and aligned with other code, WDYT?

Reply:
in this case maybe better
``
if (status == UCS_OK) {
  *ep_p = ep;
}

return status;
```